swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
+from swap_meet.item import Item
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory = inventory 
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if len(self.inventory) > 0:
+            for thing in self.inventory:
+                if thing == item:
+                    self.inventory.remove(item)
+                    return item
+        return False
+
+    def get_by_category(self,category):
+        category_list = []
+        for thing in self.inventory:
+            if thing.category == category:
+                category_list.append(thing)
+        return category_list
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.remove(my_item)
+            friend.remove(their_item)
+            friend.add(my_item)
+            self.add(their_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, friend):
+        if len(self.inventory) == 0 or len(friend.inventory) == 0:
+            return False
+        else:
+            my_item = self.inventory.pop(0)
+            friend.add(my_item)
+            their_item = friend.inventory.pop(0)
+            self.add(their_item)
```

Comment:
Could we reuse one of our methods here? Or could we do this as a big O(1) operation (recall that `pop` for an arbitrary index _must_ shift everything after that index to fill the open space, so it's O(n) at the head).

This method is an interesting look at the balance between code re-use or custom implementations that take into account additional information for better performance. We _could_ choose to reuse our previous code. _Or_ we could specialize our implementation to achieve better performance for this particular scenario.